REGRESSION(266408@main): JPEG XL does not work in STP and down levels

shallawa opened the pull request:
#### 5fb255b53fc547d77ec57e9d2d59b9cb8762d59b
<pre>
REGRESSION(266408@main): JPEG XL does not work in STP and down levels
<a href="https://bugs.webkit.org/show_bug.cgi?id=263626">https://bugs.webkit.org/show_bug.cgi?id=263626</a>
rdar://117049172

Reviewed by Alexey Proskuryakov.

Revert 266408@main to enable JPEG XL on down levels. Enable JPEG XL layout tests
on Apple ports.

Elliott Williams provided the SDKSettings changes.

* LayoutTests/fast/images/jpegxl-with-color-profile-expected.html:
* LayoutTests/fast/images/jpegxl-with-color-profile.html:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* Source/WebCore/Configurations/WebCore.xcconfig:
* WebKitLibraries/SDKs/macosx12.0-additions.sdk/SDKSettings.plist:
* WebKitLibraries/SDKs/macosx13.0-additions.sdk/SDKSettings.plist:

Canonical link: <a href="https://commits.webkit.org/269900@main">https://commits.webkit.org/269900@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bee4ffbc16ae201fd7a36dc68dd12afb39b35b03

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/23904 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/2016 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/24997 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26048 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22028 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/24179 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3629 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24396 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22529 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24148 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1561 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20662 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26637 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1313 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21572 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27804 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21790 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/21853 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25593 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1260 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/18946 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/1271 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/21346 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5736 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1671 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1595 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebCore/Configurations/WebCore.xcconfig:
```diff
@@ -26,7 +26,9 @@
-WK_USE_APPLEJPEGXL = NO;
+WK_USE_APPLEJPEGXL = $(WK_USE_APPLEJPEGXL_$(WK_PLATFORM_NAME));
+WK_USE_APPLEJPEGXL_macosx = $(WK_USE_APPLEJPEGXL$(WK_MACOS_1400));
+WK_USE_APPLEJPEGXL_MACOS_BEFORE_1400 = YES;
```
It will have to be disabled in pre-Sonoma open source builds. We include the framework in STP for older OSes, but it's not present on the bots.

emw-apple replied:
Hm…it's checked in as a SDK addition in these paths:
```
WebKitLibraries/SDKs/macosx12.0-additions.sdk/System/Library/PrivateFrameworks/AppleJPEGXL.framework/AppleJPEGXL.tbd
WebKitLibraries/SDKs/macosx13.0-additions.sdk/System/Library/PrivateFrameworks/AppleJPEGXL.framework/AppleJPEGXL.tbd
```

If I remember correctly, the idea is that open-source builds should weak-link against it and handle it being missing at runtime. That link command doesn't have the SDKAdditions path at all, so maybe there is a search path missing.

aproskuryakov replied:
I stand corrected.

shallawa committed:
REGRESSION(266408@main): JPEG XL does not work in STP and down levels
https://bugs.webkit.org/show_bug.cgi?id=263626
rdar://117049172

Reviewed by Alexey Proskuryakov.

Revert 266408@main to enable JPEG XL on down levels. Enable JPEG XL layout tests
on Apple ports.

Elliott Williams provided the SDKSettings changes.

* LayoutTests/fast/images/jpegxl-with-color-profile-expected.html:
* LayoutTests/fast/images/jpegxl-with-color-profile.html:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* Source/WebCore/Configurations/WebCore.xcconfig:
* WebKitLibraries/SDKs/macosx12.0-additions.sdk/SDKSettings.plist:
* WebKitLibraries/SDKs/macosx13.0-additions.sdk/SDKSettings.plist:

Canonical link: https://commits.webkit.org/269900@main
